solana-programs/audius_eth_registry/src/processor.rs

```diff
@@ -132,8 +329,267 @@ impl Processor {
             .map_err(|e| e.into())
     }
 
+    /// Process [ValidateMultipleSignaturesClearValidSigner]().
+    pub fn process_multiple_signatures_clear_valid_signer(
+        _program_id: &Pubkey,
+        accounts: &[AccountInfo],
+        signature_data_1: SignatureData,
+        signature_data_2: SignatureData,
+        signature_data_3: SignatureData,
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        // initialized valid signer account 1
+        let valid_signer_1_info = next_account_info(account_info_iter)?;
+        // initialized valid signer account 2
+        let valid_signer_2_info = next_account_info(account_info_iter)?;
+        // initialized valid signer account 3
+        let valid_signer_3_info = next_account_info(account_info_iter)?;
+        // signer group account
+        let signer_group_info = next_account_info(account_info_iter)?;
+        // incoming valid signer account
+        let old_valid_signer_info = next_account_info(account_info_iter)?;
+        // Sysvar Instruction account info
+        let instruction_info = next_account_info(account_info_iter)?;
+
+        // Confirm program ownership of SignerGroup and ValidSigners
+        if valid_signer_1_info.owner != _program_id
+            || valid_signer_2_info.owner != _program_id
+            || valid_signer_3_info.owner != _program_id
+            || old_valid_signer_info.owner != _program_id
+            || signer_group_info.owner != _program_id {
+            return Err(AudiusError::InvalidInstruction.into());
+        }
+
+        // clock sysvar account
+        let clock_account_info = next_account_info(account_info_iter)?;
+        let clock = Clock::from_account_info(&clock_account_info)?;
+
+        // Recover all secp instructions present in this tx
+        let instruction_recovery = Self::recover_secp_instructions(&instruction_info);
+        if instruction_recovery.is_err() {
+            return Err(AudiusError::Secp256InstructionLosing.into());
+        }
+
+        let recovered_instructions = instruction_recovery.unwrap();
+        if recovered_instructions.len() < 3 {
+            return Err(AudiusError::Secp256InstructionLosing.into());
+        }
+
+        let secp_instruction_1 = &recovered_instructions[0];
```

Comment:
While a part of the code was refactored into the loop (instruction recovery), I think the rest of the code using 1-2-3 can also become the part of this loop.